pkg/server/server.go

```diff
@@ -179,10 +179,10 @@ func (s *Server) Run() (err error) {
 			}
 
 			err := service.Run(s.context)
-			// Mark that we are in shutdown mode
-			// So no more services are started
-			s.shutdownInProgress = true
 			if err != nil {
+				// Mark that we are in shutdown mode
+				// So no more services are started
+				s.shutdownInProgress = true
```

Comment:
@sakjur Isn't there more that needs doing than just this?

Reply:
I just dived into this a little bit and it seems like this should work.
I'll check out the code and play around a little to verify, but I think this change should be mergeable.

Reply:
@sakjur So `s.shutdownInProgress` only needs to be set if an error occurred?